### PR TITLE
17016 flying settings

### PR DESCRIPTION
**Pull request by r3tk0n**
Divorces HMD flying settings from Desktop flying settings in our C++, saving them as separate values. Preserves functionality of setFlyingEnabled() to avoid breaking scripts.

While the tickbox for flying in desktop has still been removed, if the user sets flying via MyAvatar.setFlyingEnabled() in desktop mode the preference will still be respected between launches of interface.

HMD flying settings will also be respected between launches.

On first run, flying is enabled in desktop mode and disabled in HMD mode.

**TEST PLAN:**
_Goals:_
1.) Verify that flying is enabled in Desktop on first launch.
2.) Verify that flying is disabled in HMD on first launch.
3.) Verify that setting flying to enabled or disabled in HMD mode (via tablet) is respected between launches, independent of desktop mode.
4.) Verify that setting flying to enabled or disabled in Desktop mode (via MyAvatar.setFlyingEnabled() in Developer > Console) is respected between launches.

_Steps:_
- Launch interface.exe on a clean install (nothing in %appdata%) in desktop mode.
- Verify that flying is enabled. If so, Goal 1 is good.
- Switch to HMD mode with Display > (insert name of relevant HMD here)
- Verify that flying is now disabled. If so, Goal 2 is good.
- Open the tablet, go to "Menu > Settings > Control..." and enable "Flying and jumping" and "Advanced movement for hand controllers". Save these settings.
- Verify that flying is now enabled.
- Switch to Desktop mode ("Display > Desktop").
- Enable Developer menu ("Settings > Developer Menu")
- Open Console ("Developer > Console")
- Type "MyAvatar.setFlightEnabled(false)" and hit enter (should say "undefined" after enter).
- Close interface.exe and re-open in HMD mode.
- Verify that flying is still enabled in HMD. If so, Goal 3 is good.
- Switch to Desktop mode ("Display > Desktop").
- Verify that flying is still disabled in Desktop. If so, Goal 4 is good.

### CODE REVIEW

File: interface/src/avatar/MyAvatar.cpp [hunk 1]
```diff
@@ -577,9 +577,11 @@ void MyAvatar::updateChildCauterization(SpatiallyNestablePointer object, bool ca
 
 void MyAvatar::simulate(float deltaTime) {
     PerformanceTimer perfTimer("simulate");
-
+
```
Comment (wayne-chen) [inline interface/src/avatar/MyAvatar.cpp]: I think you should remove the trailing whitespace.

File: interface/src/avatar/MyAvatar.cpp [hunk 2]
```diff
@@ -2840,7 +2855,33 @@ bool MyAvatar::isInAir() {
 
 bool MyAvatar::getFlyingEnabled() {
     // May return true even if client is not allowed to fly in the zone.
-    return _enableFlying;
+    return (qApp->isHMDMode() ? getFlyingHMDPref() : getFlyingDesktopPref());
```
Comment (wayne-chen) [inline interface/src/avatar/MyAvatar.cpp]: Can we ensure that every time `getFlyingEnabled()` is called, the display plugins are already loaded? If so, I think this is fine.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11607/HighFidelity-Beta-PR13702-f3d0d9b.apk

File: interface/src/avatar/MyAvatar.cpp (hunk 2 above)
Comment (r3tk0n) [reply interface/src/avatar/MyAvatar.cpp]: Aye, it's actually called very rarely in the code,. There are calls to isHMDMode() in Application.cpp that are called well before this.

File: interface/src/avatar/MyAvatar.cpp (hunk 1 above)
Comment (r3tk0n) [reply interface/src/avatar/MyAvatar.cpp]: I was following the style of other items in that area beforehand, figured it best to segregate unrelated calls.

File: interface/src/avatar/MyAvatar.cpp (hunk 2 above)
Comment (wayne-chen) [reply interface/src/avatar/MyAvatar.cpp]: Okay, thanks for the clarification.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11607/HighFidelity-Beta-PR13702-f3d0d9b.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11607/HighFidelity-Beta-Interface-PR13702-f3d0d9b.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11607/HighFidelity-Beta-PR13702-f3d0d9b.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11607/HighFidelity-Beta-Interface-PR13702-f3d0d9b.exe

**Comment by qliejahner** [comment]
PASS @conklin94122 

By default, flight is enabled for desktop and disabled for HMD, and changes in settings for one view doesn't affect the other view.

**Pull request merged**